code/go/0chain.net/smartcontract/minersc/fees_test.go

```diff
@@ -1,483 +1,487 @@
+			for _, staker := range sharder.stakers {
+				var _, err = staker.callAddToDelegatePool(t, msc, now,
+					SharderStakeValue, sharder.client.id, balances)
+
+				require.NoError(t, err, "staking sharder")
+				now += timeDelta
+			}
+		}
+
+		balances.requireNodesHaveZeros(t, sharders,
+			"sharders' balances must be unchanged so far")
+		balances.requireStakersHaveZeros(t, sharders,
+			"stakers' balances must be unchanged so far")
+    })
+
+
+	msc.setDKGMiners(t, miners, balances)
+
+
+	t.Run("pay fees [vc, no fees]", func(t *testing.T) {
+		balances.requireAllBeZeros(t)
+		msc.setRounds(t, 250, 251, balances)
+
+		msc.requirePendingPoolsBeNotEmpty(t, balances)
+		msc.requireActivePoolsBeEmpty(t, balances)
+
+		setMagicBlock(t,
+			unwrapClients(miners),
+			unwrapClients(sharders),
+			balances)
+
+		now += timeDelta
+		// all rewards go the nodes
+		// nothing must be paid to stakers, but pools become active
+		msc.callPayFees(t, balances, miners, sharders,
+			generator.client.id, 0, 251, now)
+
+		msc.requireActivePoolsBeNotEmpty(t, balances)
+
+		balances.requireTotalAmountBeEqual(t, BlockReward)
+		balances.requireStakersHaveZeros(t, miners, "stakers' balances must be unchanged so far")
+		balances.requireStakersHaveZeros(t, sharders, "stakers' balances must be unchanged so far")
+
+		var global = msc.retrieveGlobalNode(t, balances)
+		require.EqualValues(t, 251, global.LastRound)
+
+		msc.verifyRewards(t, false, false,
+			balances, generator, miners, sharders)
+
+		balances.requireTotalAmountBeEqual(t, BlockReward)
+	})
+
+	msc.setDKGMiners(t, miners, balances)
+
+	t.Run("pay fees [no vc, no fees]", func(t *testing.T) {
+		msc.resetBalancesAndRewards(balances)
+		msc.setRounds(t, 251, 501, balances)
+
+		msc.requirePendingPoolsBeEmpty(t, balances)
+		msc.requireActivePoolsBeNotEmpty(t, balances)
+
+		now += timeDelta
+		// pools active, no fees, part of rewards must be payed
+		// to generator's and block sharders' stake holders
+		msc.callPayFees(t, balances, miners, sharders,
+			generator.client.id, 0, 252, now)
+
+		balances.requireTotalAmountBeEqual(t, BlockReward)
+
+		msc.verifyRewards(t, true, false,
+			balances, generator, miners, sharders)
+	})
+
+	// don't set DKG miners list, because no VC is expected
+
+	t.Run("pay fees [no vc, fees]", func(t *testing.T) {
+		msc.resetBalancesAndRewards(balances)
+		msc.setRounds(t, 252, 501, balances)
+
+		msc.requirePendingPoolsBeEmpty(t, balances)
+		msc.requireActivePoolsBeNotEmpty(t, balances)
+
+		now += timeDelta
+		// pools are active, rewards as above and +fees
+		msc.callPayFees(t, balances, miners, sharders,
+			generator.client.id, TransactionFee, 253, now)
+
+		balances.requireTotalAmountBeEqual(t, BlockReward + TransactionFee)
+
+		msc.verifyRewards(t, true, true,
+			balances, generator, miners, sharders)
+	})
+
+	// don't set DKG miners list, because no VC is expected
+
+	t.Run("pay fees [vc, interests]", func(t *testing.T) {
+		msc.resetBalancesAndRewards(balances)
```

Comment:
This failed when I tried it
```
=== RUN   Test_payFees/pay_fees_[vc,_interests]
    require.go:157: 
        	Error Trace:	fees_test.go:237
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	Test_payFees/pay_fees_[vc,_interests]
        	Messages:   	TODO
    --- FAIL: Test_payFees/pay_fees_[vc,_interests] (11.55s)
```

Reply:
yes, this is stub for another PR, only plain fees here